lib/karafka/params/params.rb

```diff
@@ -48,20 +48,21 @@ def deserialize!
         return self if self['deserialized']
 
         self['deserialized'] = true
-        self['payload'] = deserialize(self['payload'])
+        self['payload'] = deserialize(self)
         self
       end
 
       private
 
-      # @param payload [String] Raw data that we want to deserialize using consumer deserializer
+      # @param params [Karafka::Params::Params] Full params object that we want to
+      # deserialize using consumer deserializer
       # @return [Object] deserialized data
-      def deserialize(payload)
-        Karafka.monitor.instrument('params.params.deserialize', caller: self) do
-          self['deserializer'].call(payload)
+      def deserialize(params)
+        Karafka.monitor.instrument('params.params.deserialize', caller: params) do
+          params['deserializer'].call(params)
```

Comment:
As you are passing `Karafka::Params::Params` object now, we can use `params.deserializer`